src/components/login/login.component.js

```diff
@@ -1,43 +1,48 @@
+import { bindActionCreators } from 'redux';
+import loginUserAction from './login.actions'
+import { connect } from 'react-redux';
+
 
 const initialValues = {
     email: '',
-    password: '' 
+    password: ''
 }
 
 const LoginSchema = Yup.object().shape({
     email: Yup.string()
-    .email("Invalid email"),
+        .email("Invalid email"),
     password: Yup.string()
-    .min(5, "Too Short!")
-    .max(50, "Too Long!")
+        .min(5, "Too Short!")
+        .max(50, "Too Long!")
 });
 
-const Login = () => (
+const LoginPage = (props) => (
     <Formik
-    initialValues = {initialValues}
-    validationSchema={LoginSchema}
-    render={({ errors, touched }) => (
-        <Form>
-        <div className="form-group">
-            <label htmlFor="email">Email</label>
-            <Field name="email" type="text" className={'form-control' + (errors.email && touched.email ? ' is-invalid' : '')} />
-            <ErrorMessage name="email" component="div" className="invalid-feedback" />
-        </div>
-        <div className="form-group">
-            <label htmlFor="password">Password</label>
-            <Field name="password" type="password" className={'form-control' + (errors.password && touched.password ? ' is-invalid' : '')} />
-            <ErrorMessage name="password" component="div" className="invalid-feedback" />
-        </div>
-        <div className="form-group">
-            <button type="submit" className="btn btn-primary mr-2">Login</button>
-            <button type="button" className="btn btn-secondary">Signup</button>
-        </div>
-    </Form>
-    )} />
+        initialValues={initialValues}
+        validationSchema={LoginSchema}
+        onSubmit={(values) => (props.loginUserAction(values))}
+        render={({ errors, touched }) => (
+            <Form>
+                <div className="form-group">
+                    <label htmlFor="email">Email</label>
+                    <Field name="email" type="text" className={'form-control' + (errors.email && touched.email ? ' is-invalid' : '')} />
+                    <ErrorMessage name="email" component="div" className="invalid-feedback" />
+                </div>
+                <div className="form-group">
+                    <label htmlFor="password">Password</label>
+                    <Field name="password" type="password" className={'form-control' + (errors.password && touched.password ? ' is-invalid' : '')} />
+                    <ErrorMessage name="password" component="div" className="invalid-feedback" />
+                </div>
+                <div className="form-group">
+                    <button type="submit" className="btn btn-primary mr-2">Login</button>
+                    <button type="button" className="btn btn-secondary">Signup</button>
+                </div>
+            </Form>
+        )} />
 );
 
-
-export default Login;
+const mapDispatchToProps = dispatch => (bindActionCreators({loginUserAction}, dispatch))
```

Comment:
i think this should be in a container

Reply:
So any wrapper that produces a higher-level component should go in container.js? 
For ex. using withFormik(LoginPage)

Reply:
anything that involves redux or logic i.e. props, actions, handlers

Reply:
ideally yeah, Formik stuff should be in a container since it's logic